moto/firehose/models.py

```diff
@@ -0,0 +1,644 @@
+        )
+
+        if delivery_stream_name in self.delivery_streams:
+            raise ResourceInUseException(
+                f"Firehose {delivery_stream_name} under accountId {ACCOUNT_ID} "
+                f"already exists"
+            )
+
+        if len(self.delivery_streams) == DeliveryStream.MAX_STREAMS_PER_REGION:
+            raise LimitExceededException(
+                f"You have already consumed your firehose quota of "
+                f"{DeliveryStream.MAX_STREAMS_PER_REGION} hoses. Firehose "
+                f"names: {list(self.delivery_streams.keys())}"
+            )
+
+        # Rule out situations that are not yet implemented.
+        if delivery_stream_encryption_configuration_input:
+            raise NotImplementedError(
+                "A delivery stream with server-side encryption enabled is not "
+                "yet implemented"
+            )
+
+        if destination_name == "Splunk":
+            raise NotImplementedError(
+                "A Splunk destination delivery stream is not yet implemented"
+            )
+
+        if (
+            kinesis_stream_source_configuration
+            and delivery_stream_type != "KinesisStreamAsSource"
+        ):
+            raise InvalidArgumentException(
+                "KinesisSourceStreamConfig is only applicable for "
+                "KinesisStreamAsSource stream type"
+            )
+
+        # Validate the tags before proceeding.
+        errmsg = self.tagger.validate_tags(tags or [])
+        if errmsg:
+            raise ValidationException(errmsg)
+
+        # Create a DeliveryStream instance that will be stored and indexed
+        # by delivery stream name.  This instance will update the state and
+        # create the ARN.
+        delivery_stream = DeliveryStream(
+            region,
+            delivery_stream_name,
+            delivery_stream_type,
+            kinesis_stream_source_configuration,
+            destination_name,
+            destination_config,
+        )
+        self.tagger.tag_resource(delivery_stream.delivery_stream_arn, tags or [])
+
+        self.delivery_streams[delivery_stream_name] = delivery_stream
+        return self.delivery_streams[delivery_stream_name].delivery_stream_arn
+
+    def delete_delivery_stream(
+        self, delivery_stream_name, allow_force_delete=False
+    ):  # pylint: disable=unused-argument
+        """Delete a delivery stream and its data.
+
+        AllowForceDelete option is ignored as we only superficially
+        apply state.
+        """
+        delivery_stream = self.delivery_streams.get(delivery_stream_name)
+        if not delivery_stream:
+            raise ResourceNotFoundException(
+                f"Firehose {delivery_stream_name} under account {ACCOUNT_ID} "
+                f"not found."
+            )
+
+        self.tagger.delete_all_tags_for_resource(delivery_stream.delivery_stream_arn)
+
+        # The following logic is not applicable for moto as far as I can tell.
```

Comment:
Agreed - I think we can just delete the logic, no need to have it commented out

Reply:
Done.